VS-9: Patient data is not deleted from VariantStore when patient record is deleted

veronikaslc opened the pull request.

veronikaslc committed:
[misc] Add getAllVariantsForIndividual(id) functionality

veronikaslc committed:
[misc] Refactoring removeIndividual(id) code

veronikaslc committed:
[misc] Updating VCFRemovalJob

veronikaslc committed:
[misc] Addressing codereview comments

veronikaslc committed:
[misc] Adding parameters check to VariantStoreUploadService

veronikaslc committed:
[misc] Updating readme

buske commented on integration/api/src/main/java/org/phenotips/variantStoreIntegration/internal/DefaultVCFUploadManager.java:
```diff
@@ -144,9 +144,10 @@ public void uploadVCF(String patientID, String filePath) throws Exception
             this.logger.warn("Variant store exception thrown when trying to upload a vcf for: {}", patientID);
-            e.printStackTrace();
+            //e.printStackTrace();
+            throw new Exception("Variant store exception thrown when trying to upload a vcf for: "
```
Why do you raise a new, generic `Exception` instead of re-raising the same one or not catching it at all, if you're not actually handling it here?

buske commented on integration/api/src/main/java/org/phenotips/variantStoreIntegration/internal/DefaultVCFUploadManager.java:
```diff
@@ -176,27 +177,36 @@ public void cancelUpload(Patient patient)
-            e.printStackTrace();
+            this.logger.warn("Variant store exception thrown when trying to remove a vcf for: {}", patientID);
+            //e.printStackTrace();
+            throw new Exception("Variant store exception thrown when trying to remove a vcf for: "
```
Same question here...

veronikaslc committed:
VS-9: Patient data is not deleted from VariantStore when patient record is deleted

Removing unnecessary exceptions thrown

Commit:
PT-2700: Increment PhenoTips version and make the patient-data APIs extend the new entities API

Extending d13f25cd74e51e56aaad8e24b002ffd131530c42 by @veronikaslc to variant-store

Commit:
[misc] Fixed wrong StringUtils import
